sdk/cosmos/azure-cosmos-encryption/src/main/java/com/azure/cosmos/encryption/CosmosEncryptionAsyncContainer.java

```diff
@@ -193,7 +201,93 @@ public Mono<CosmosItemResponse<Object>> deleteItem(String itemId,
                                                        PartitionKey partitionKey,
                                                        CosmosItemRequestOptions requestOptions) {
 
-        return container.deleteItem(itemId, partitionKey, requestOptions);
+        return deleteItemInternal(itemId, partitionKey, requestOptions);
+    }
+
+    private Mono<CosmosItemResponse<Object>> deleteItemInternal(String itemId, PartitionKey partitionKey, CosmosItemRequestOptions requestOptions) {
+        this.encryptionProcessor.initEncryptionSettingsIfNotInitializedAsync();
+        return Mono.just(this.encryptionProcessor.getEncryptionSettings())
+            .flatMap(settings -> {
+                try {
+                    return Mono.zip(
+                        checkAndGetEncryptedId(itemId, settings),
+                        checkAndGetEncryptedPartitionKey(partitionKey, settings)
+                    ).flatMap(encryptedIdPartitionTuple -> container.deleteItem(encryptedIdPartitionTuple.getT1(), encryptedIdPartitionTuple.getT2(), requestOptions));
+                } catch (Exception ex) {
```

Comment:
what are we catching here? who is throwing the exception?

Reply:
Removed the exception, thanks for pointing that out.